service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/BlobCopierConnectedTest.java

```diff
@@ -32,11 +31,11 @@
 import java.util.List;
 import java.util.UUID;
 import java.util.stream.Collectors;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.Tag;
-import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.TestInstance;
+import javax.annotation.Nullable;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 import org.springframework.beans.factory.annotation.Autowired;
 
 @Tag("azureConnectedPlus")
```

Comment:
This test doesn't run on PR commits, but I ran it locally and [here](https://github.com/DataBiosphere/terra-workspace-manager/actions/runs/4628382886) (failure was a sporadic unrelated test failure… rerunning just to be sure).

Reply:
I know it can be a lot of obnoxious setup at first, with how some of this code is structured.... but might it be a good time to break some of this out into a proper unit test?

Reply:
I don't think so because this is the only coverage that we have of actually using the Azure APIs to copy blobs. I found it really useful to be able to see what Azure would return when we had "directory" structures, and if it would prefix blob names with "/", etc.

Reply:
Yes, this test is intended to be a real (limited) exercise of our ability to copy blobs.

Reply:
I wasn't necessarily saying that what's here has to go away (particularly since it doesn't run on PRs).

That said, I realize there isn't a ton of logic in the BlobCopier class in the first place, so the answer might be that it's not worth it atm.